code/game/machinery/mind_machine.dm

```diff
@@ -0,0 +1,846 @@
+			spawn(5)
+				delayedSwap()
+				return TRUE
+
+//////Scan and Swap, other UI procs//////////////
+
+/obj/machinery/mind_machine/mind_machine_hub/proc/unlockPods()
+	occupantScan = FALSE
+	mindTypeOne = "None"
+	mindTypeTwo = "None"
+	occupantNameOne = "None"
+	occupantNameTwo = "None"
+	occupantStatOne = "None"
+	occupantStatTwo = "None"
+	connectOne.icon_state = "mind_pod_open"
+	connectTwo.icon_state = "mind_pod_open"
+	playsound(connectTwo, 'sound/machines/door_unbolt.ogg', 35, 1)
+	playsound(connectOne, 'sound/machines/door_unbolt.ogg', 35, 1)
+	if(badSwap == FALSE)
+		flick("mind_pod_opening", connectOne)
+		flick("mind_pod_opening", connectTwo)
+	else
+		flick("mind_pod_thefly", connectOne)
+		flick("mind_pod_thefly", connectTwo)
+		badSwap = FALSE
+	lockedPods = FALSE
+
+/obj/machinery/mind_machine/mind_machine_hub/proc/ejectPods()
+	connectOne.go_out()
+	connectTwo.go_out()
+
+/obj/machinery/mind_machine/mind_machine_hub/proc/delayedSwap()
+	if(!occupantScan)
+		errorMessage = "Proceed with caution: supervisor advised"
+		sleep(5 SECONDS)
+		currentlySwapping = FALSE
+		if((occupantOne) && (occupantTwo))
+			scanOccupants()
+			if(prob(8-manipRating)) //A tiny bit risky with no upgrades, 0 risk with full xenoarch manipulators.
+				malfSwap = TRUE
+			swapOccupants()
+			unlockPods()
+		else
+			errorMessage = "Missing occupant"
+
+/obj/machinery/mind_machine/mind_machine_hub/proc/scanOccupants(var/mob/S)
+	if((!locate(occupantOne,connectOne.contents)) || (!locate(occupantTwo, connectTwo.contents)))
+		errorMessage = "Missing occupant"
+		occupantOne = null
+		occupantTwo = null
+		return
+	currentlySwapping = TRUE
+	lockedPods = TRUE
+	if((connectOne.icon_state != "mind_pod_closed") && (connectTwo.icon_state != "mind_pod_closed"))
+		playsound(connectOne, 'sound/machines/poddoor.ogg', 55, 1)
+		playsound(connectTwo, 'sound/machines/poddoor.ogg', 55, 1)
+		flick("mind_pod_closing",connectOne)
+		connectOne.icon_state = "mind_pod_closed"
+		flick("mind_pod_closing",connectTwo)
+		connectTwo.icon_state = "mind_pod_closed"
+	sleep(2 SECONDS)
+	playsound(connectOne, 'sound/effects/sparks4.ogg', 80, 1)
+	playsound(connectTwo, 'sound/effects/sparks4.ogg', 80, 1)
+	occupantScan = TRUE
+	S = occupantOne
+	occupantNameOne = S.name
+	switch(S.stat)
+		if(0)
+			occupantStatOne = "Alive"
+		if(1)
+			occupantStatOne = "Unconscious"
+		if(2)
+			occupantStatOne = "Dead"
+	if(!S.mind)
+		mindTypeOne = "Lower" //Simple mob
+	if(S.mind)
+		if(isrobot(S))
+			mindTypeOne = "Artificial" //Silicon player, obviously
+		else
+			mindTypeOne = "Higher" //Player controlled
+	if(isvampire(S) || isanycultist(S) || ischangeling(S) || ismalf(S))
+		mindTypeOne = "Shielded" //Mostly to fix spell bugs but also tinfoil
+	if((ishigherbeing(S)) || (ismonkey(S)))
+		var/mob/living/carbon/T = S
+		if(T.is_wearing_item(/obj/item/clothing/head/tinfoil))
+			mindTypeTwo = "Shielded"
+	S = occupantTwo
+	occupantNameTwo = S.name //and for pod two
+	switch(S.stat)
```

Comment:
This is a carbon copy of the thing for occupant one. Please make this a proc so that you don't need to copypaste as much.